android/app/src/main/java/org/fossasia/openevent/activities/SessionDetailActivity.java

```diff
@@ -358,16 +358,26 @@ public boolean onOptionsItemSelected(MenuItem item) {
                 return true;
 
             case R.id.action_share:
+                StringBuilder speakersNameBuilder  = new StringBuilder();
+                for(Speaker speaker : session.getSpeakers()){
+                    if(speakersNameBuilder.toString().equals("")){
```

Comment:
This is O(n) for each operation. So the whole ```for``` loop will be O(n^2) which is bad. Please use a boolean flag.

Reply:
@srv-twry Using a boolean flag might involve another integer variable and `.size()` method too. Is it still advisable?

Reply:
It seems you haven't understood what i am suggesting. So here is the full implementation.

``` Java
boolean first = true;
for(Speaker speaker : session.getSpeakers()){
    if(!first) {
         speakersNameBuilder.append(", ");
    }
    speakersNameBuilder.append(speaker.getName());
    first = false;
}
```

Ignore indentation errors here.

Reply:
@srv-twry Wow! That was neat! 😅